src/dbnode/storage/index/types.go

```diff
@@ -155,6 +155,9 @@ type QueryResultsOptions struct {
 	// SizeLimit will limit the total results set to a given limit and if
 	// overflown will return early successfully.
 	SizeLimit int
+
+	// FilterID if not nil will filter the ID if specified.
+	FilterID func(id ident.ID) bool
```

Comment:
"FilterID, if provided, can be used to filter out unwanted IDs from the query results"

Maybe just also say we currently use it for filtering by shard?